File: packages/native/modules/cue-native/ios/CueLegacyPreferencesModule.swift
import ExpoModulesCore

/// Reads what the Capacitor build left in `UserDefaults`.
///
/// Capacitor Preferences namespaces by a group that defaults to
/// `CapacitorStorage`, and on iOS that group plus a dot is prefixed onto every
/// key, so the stored name is `CapacitorStorage.cue.trakt.token`. Android puts
/// the group in the file name and leaves the key alone. The two shapes are not
/// the same, and a migration that assumes one reads nothing on the other
/// platform, so the prefix is applied here rather than in shared code.
///
/// React Native's own `Settings` module wraps `NSUserDefaults` and is documented
/// as iOS-only, which would cover half the job; Android needs a native module
/// regardless, so both platforms go through this one.
public class CueLegacyPreferencesModule: Module {
    private static let prefix = "CapacitorStorage."

    public func definition() -> ModuleDefinition {
        Name("CueLegacyPreferences")

        AsyncFunction("read") { (key: String) -> String? in
            UserDefaults.standard.string(forKey: Self.prefix + key)
        }

        AsyncFunction("remove") { (key: String) in
            UserDefaults.standard.removeObject(forKey: Self.prefix + key)
        }
    }
}

File: packages/native/modules/cue-native/android/src/main/java/app/cuetracker/cuenative/CueLegacyPreferencesModule.kt
package app.cuetracker.cuenative

import android.content.Context
import expo.modules.kotlin.exception.Exceptions
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Reads what the Capacitor build left in shared preferences.
 *
 * Capacitor Preferences namespaces by a group that defaults to
 * `CapacitorStorage`, and on Android that group is the file name while the key
 * is stored unprefixed. iOS prefixes the group onto the key inside
 * `UserDefaults` instead. The two shapes are not the same, and a migration that
 * assumes one reads nothing on the other platform, so the difference is
 * expressed here rather than in shared code.
 */
class CueLegacyPreferencesModule : Module() {
    private companion object {
        const val GROUP = "CapacitorStorage"
    }

    override fun definition() = ModuleDefinition {
        Name("CueLegacyPreferences")

        AsyncFunction("read") { key: String ->
            preferences().getString(key, null)
        }

        AsyncFunction("remove") { key: String ->
            preferences().edit().remove(key).apply()
        }
    }

    private fun preferences() =
        (appContext.reactContext ?: throw Exceptions.ReactContextLost())
            .getSharedPreferences(GROUP, Context.MODE_PRIVATE)
}
